src/shared/fs/watchedFiles.ts

```diff
@@ -275,26 +282,37 @@ export abstract class WatchedFiles<T> implements vscode.Disposable {
     }
 
     /**
-     * Rebuilds registry using current glob and exclusion patterns.
-     * All functionality is currently internal to class, but can be made public if we want a manual "refresh" button
+     * Builds/rebuilds registry using current glob and exclusion patterns. ***Necessary to init registry***.
+     *
+     * @param cancellationTimeout Optional timeout that can trigger canceling additional loading on completion (manual or timed)
      */
-    public async rebuild(): Promise<void> {
+    public async rebuild(cancellationTimeout?: Timeout): Promise<void> {
+        let skips = 0
+        this.isWatching = true
         this.reset()
 
         const exclude = getExcludePatternOnce()
-        for (const glob of this.globs) {
+        getLogger().info(`${this.name}: Building registry with the following criteria: ${this.outputPatterns()}`)
+        for (let i = 0; i < this.globs.length && !cancellationTimeout?.completed; i++) {
+            const glob = this.globs[i]
             try {
                 const found = await vscode.workspace.findFiles(glob, exclude)
-                for (const item of found) {
-                    await this.addItem(item, true)
+                for (let j = 0; j < found.length && !cancellationTimeout?.completed; j++) {
+                    await this.addItem(found[j], true)
                 }
             } catch (e) {
+                // file not processable
+                skips++
                 const err = e as Error
-                if (err.name !== 'Canceled') {
```

Comment:
why remove this? I believe I added this because vscode can throw nonsense "Canceled" errors when shutting down the debugger. So we don't want to log those because they are misleading.